app/models/processor.rb

```diff
@@ -0,0 +1,81 @@
+require 'json'
+require 'useragent'
+require 'uri'
+
+module Processor
+extend self
+
+  def clean_data(params)
+    {identifier: params[:identifier], root_url: params[:rootUrl]}
+  end
+
+  def json_parser(params)
+    JSON.parse(params)
+  end
+
+  def params_cleaner(params)
+    user_agent, referredby, url = params_parser(params)
+
+    {
+     requested_at: json_parser(params)['requestedAt'],
+     responded_in: json_parser(params)['respondedIn'],
+     referred_by_id: ReferredBy.find_or_create_by(root_url: referredby.host, path: referredby.path).id,
+     request_type_id: RequestType.find_or_create_by(http_verb: json_parser(params)['requestType']).id,
```

Comment:
so rather than setting the foreign_key's here, meaning `request_type_id`, `event_id`, etc. we could set the objects - `request_type` and `event` equal to the actual object rather than the id. it would look something like this...
```ruby
...
request_type: RequestType.find_or_create_by(http_verb: json_parser(params)['requestType']),
event: Event.find_or_create_by(event_name: json_parser(params)['eventName'])
...
```

whenever possible, we should avoid setting foreign keys and let ActiveRecord help us out by sending objects rather than specific id's